### PR TITLE
Added local management support in e2e tests

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Added support for local management in e2e test application
* Added `LocalManagementTest` to perform Set via local management using CommandRunner
* Enabled local management `"LocalManagement": 1` on test targets
* Added additional verbosity to e2e logs
* Added `ValidateLocalReported` to test local reported objects are in sync

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/workflows/e2etest-run-tests.yml
Comment (robertschaedler3) [inline .github/workflows/e2etest-run-tests.yml:116]: Just curious, why move these environment variables out of `env` ?

**Review by robertschaedler3 — APPROVED** [review]
👍🏻

File: src/tests/e2etest/LocalManagementTests.cs
Comment (MariusNi) [inline src/tests/e2etest/LocalManagementTests.cs:59]: >// TODO: try setting [](http://example.com/codeflow?start=12&length=20)

Don't merge this TODO in. File a bug or ignore and either way remove the TODO in comment

File: src/tests/e2etest/E2ETest.cs
Comment (MariusNi) [inline src/tests/e2etest/E2ETest.cs:167]: Local management is by default disabled. Is the E2E test enabling that and where?

Comment (MariusNi) [reply src/tests/e2etest/E2ETest.cs:167]: Ok, found where in the yml

Comment (MariusNi) [inline src/tests/e2etest/E2ETest.cs:167]: > protected void ValidateLocalReported(Object reported, String componentName, String objectName = "") [](http://example.com/codeflow?start=7&length=100)

This PR appears to only add coverage for the RC path. The DC path is not covered at all apparently. Is this inaccurate?

Two asks:

1. Need to validate that when something is changed via either DC or Desired Twin that both RC and Reported Twin are updated.

2. Need to validate that when something is changed first from DC and then Twin or vice versa that the RC and Reported Twin contain last respectively.

File: src/tests/e2etest/E2ETest.cs
Comment (MariusNi) [inline src/tests/e2etest/E2ETest.cs:26]: This is for RC only. We also need the DC path validated (see the other comment)

File: src/tests/e2etest/LocalManagementTests.cs
Comment (MariusNi) [inline src/tests/e2etest/LocalManagementTests.cs:29]: Not enough for DC path. Both RC and DC paths must be tested same as Reported and Desired Twin paths, for each and every module, 100%.

File: .github/workflows/e2etest-run-tests.yml
Comment (AhmedBM) [reply .github/workflows/e2etest-run-tests.yml]: because env only give the current users environment variables, we require them with our privileged user now because we need write access to the local desired json.

File: devops/e2e/ubuntu-focal-arm64-cloud-init.tpl
Comment (MariusNi) [inline devops/e2e/ubuntu-focal-arm64-cloud-init.tpl:24]: >permissions: "0644" [](http://example.com/codeflow?start=4&length=19)

Would be nice to add a comment that describes what permissions are

**Review by MariusNi — DISMISSED** [review]
:shipit:

File: src/tests/e2etest/CommandRunnerTests.cs
Comment (robertschaedler3) [inline src/tests/e2etest/CommandRunnerTests.cs:63]: Is there a way to keep these to be `PascalCased`? There was a change a while back that fixed this: #181

Comment (AhmedBM) [reply src/tests/e2etest/CommandRunnerTests.cs:63]: oups, yes, i saw that. I didn't intend to have those changes in there. I will change it back.

**Review by robertschaedler3 — APPROVED** [review]
👍🏻

**Pull request merged**
